notifications/v1/client.go

```diff
@@ -0,0 +1,95 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/sourcegraph/log"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// Client is a client for receiving SAMS notifications from a GCP Pub/Sub
+// subscription.
+type Client struct {
+	logger        log.Logger
+	subscription  *pubsub.Subscription
+	cancelContext context.CancelFunc
+}
+
+// NewClient creates a new Client for receiving SAMS notifications with given
+// GCP project ID and Pub/Sub subscription ID.
+func NewClient(logger log.Logger, projectID, subscriptionID string) (*Client, error) {
```

Comment:
@unknwon we should offer standardized constructor-from-env setup like #15 for this

Reply:
Please see my comment in https://github.com/sourcegraph/sourcegraph-accounts-sdk-go/pull/24#issuecomment-2100712267.

>Or are these things I would need to wire into my MSP service configuration?

TLDR yes, I think things would be clearer to be documented in the upcoming integration docs than docstring here (because here it missing the context of provisioning the actual Pub/Sub subscription).

LMK what do you think, can be convinced 😁

Reply:
> maybe we should offer standardized constructor-from-env setup like #15 for this?

👍 good idea, will push a commit.